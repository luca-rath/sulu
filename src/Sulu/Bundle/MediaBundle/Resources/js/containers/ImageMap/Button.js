// @flow
import React from 'react';
import {Icon} from 'sulu-admin-bundle/components';
import buttonStyles from './button.scss';

type Props = {
    disabled: boolean,
    icon: string,
    onClick: () => void,
};

class Button extends React.PureComponent<Props> {
    static defaultProps = {
        disabled: false,
    };

    handleClick = (event: SyntheticEvent<HTMLButtonElement>) => {
        const {onClick} = this.props;

        event.preventDefault();
        onClick();
    };

    render() {
        const {disabled, icon} = this.props;

        return (
            <button
                className={buttonStyles.button}
                disabled={disabled}
                onClick={this.handleClick}
                type="button"
            >
                <Icon name={icon} />
            </button>
        );
    }
}

export default Button;
