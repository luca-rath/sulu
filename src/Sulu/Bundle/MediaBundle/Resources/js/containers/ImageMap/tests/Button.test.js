// @flow
import React from 'react';
import {render, shallow} from 'enzyme';
import Button from '../Button';

test('Should render with icon', () => {
    expect(render(<Button icon="su-plus-circle" onClick={jest.fn()} />)).toMatchSnapshot();
});

test('Should render with icon and disabled', () => {
    expect(render(<Button disabled={true} icon="su-plus-circle" onClick={jest.fn()} />)).toMatchSnapshot();
});

test('Should call the callback on click', () => {
    const preventDefaultSpy = jest.fn();
    const onClick = jest.fn();
    const button = shallow(<Button icon="su-plus-circle" onClick={onClick} />);
    button.find('button').simulate('click', {preventDefault: preventDefaultSpy});
    expect(preventDefaultSpy).toBeCalled();
    expect(onClick).toBeCalled();
});
