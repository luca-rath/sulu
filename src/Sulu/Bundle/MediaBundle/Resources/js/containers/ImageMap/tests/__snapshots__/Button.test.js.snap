// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render with icon 1`] = `
<button
  class="button"
  type="button"
>
  <span
    aria-label="su-plus-circle"
    class="su-plus-circle"
  />
</button>
`;

exports[`Should render with icon and disabled 1`] = `
<button
  class="button"
  disabled=""
  type="button"
>
  <span
    aria-label="su-plus-circle"
    class="su-plus-circle"
  />
</button>
`;
